facia-press/app/frontpress/FapiFrontPress.scala

```diff
@@ -52,7 +49,11 @@ object LiveFapiFrontPress extends FapiFrontPress {
   def pressByPathId(path: String): Future[Unit] =
     getPressedFrontForPath(path)
       .map { pressedFront => S3FrontsApi.putLiveFapiPressedJson(path, Json.stringify(Json.toJson(pressedFront)))}
-      .asFuture.map(_.fold(e => throw new RuntimeException(s"${e.cause} ${e.message}"), _ => ()))
+      .asFuture.map(_.fold(
+        e => {
```

Comment:
I think it's fine as it is. This isn't a `Future`, rather an `Either`. With `onSuccess` you would just be using a `Success` with another nested match inside for the `Either.` The fold is to help avoid all the boilerplate.

If you want a maintain readability then just use the `map` with a match on `Left` and `Right`.